contexts-state-store/store/context_config_store.go

```diff
@@ -0,0 +1,39 @@
+package store
+
+import (
+	"github.com/kurtosis-tech/kurtosis/contexts-state-store/api/golang/generated"
+	"github.com/kurtosis-tech/kurtosis/contexts-state-store/store/persistence"
+	"sync"
+)
+
+var (
+	once               sync.Once
+	contextConfigStore ContextConfigStore
+)
+
+type ContextConfigStore interface {
+	// GetKurtosisContextsState returns the list of currently saved contexts.
+	GetKurtosisContextsState() (*generated.KurtosisContextsState, error)
+
+	// GetCurrentContext returns the current context information.
+	GetCurrentContext() (*generated.KurtosisContext, error)
+
+	// SwitchContext switches to the context passed as an argument.
+	// It throws an error if the contextUuid does not point to any known context.
+	SwitchContext(contextUuid *generated.ContextUuid) error
+
+	// AddNewContext adds a new context to the store.
+	// It throws an error if a context with the same UUID already exists
+	AddNewContext(newContext *generated.KurtosisContext) error
```

Comment:
should the store be providing the id or should the user?

Reply:
Probably neither. In the real world, to connect to a remote context, the connection file will provide it. Once we start supporting multiple local context, the CLI can generate one on the fly if it helps. In any case, I'd like to restrict the store to its only purpose: storing. At least for now, if that works for you